core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSequencerPartitionContainer.java

```diff
@@ -207,8 +206,13 @@ public Run getLastRun() {
   }
 
   @Override
-  public void addRun(Run run) {
-    runs.add(run);
+  public void setRun(Run run) {
+    if (runs.size() < 2) {
+      runs.add(run);
```

Comment:
This can result in NPE and also acts as an add instead of a set. After setRun(Run notNull) is called, `runs.size() == 1` should always be true. Change to something like:
```
if (runs != null && runs.size() > 1) {
  throw new IllegalStateException("Cannot set single run on container with multiple runs already linked!");
} else {
  runs = new ArrayList<>();
  if (run != null) runs.add(run);
}
```